src/containerapp/azext_containerapp/tests/latest/test_containerapp_sessionpool.py

```diff
@@ -181,3 +181,68 @@ def test_containerapp_sessionpool_registry(self, resource_group):
         sessionpool_list = self.cmd("containerapp sessionpool list -g {}".format(resource_group)).get_output_in_json()
         self.assertTrue(len(sessionpool_list) == 0)
 
+    @AllowLargeResponse(8192)
+    @ResourceGroupPreparer()
+    def test_containerapp_sessionpool_registry_identity(self, resource_group):
+        location = 'eastasia'
+        self.cmd('configure --defaults location={}'.format(location))
+
+        user_identity_name = self.create_random_name(prefix='sp-msi1', length=24)
+        identity_json = self.cmd('identity create -g {} -n {}'.format(resource_group, user_identity_name)).get_output_in_json()
+        user_identity_id = identity_json["id"]
+        principal_id = identity_json["principalId"]
+        
+        env_name = self.create_random_name(prefix='aca-sp-env-registry', length=24)
+        self.cmd('containerapp env create -g {} -n {} --location {} --logs-destination none'.format(resource_group, env_name, location), expect_failure=False)
+        containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+        while containerapp_env["properties"]["provisioningState"].lower() in ["waiting", "inprogress"]:
+            time.sleep(5)
+            containerapp_env = self.cmd('containerapp env show -g {} -n {}'.format(resource_group, env)).get_output_in_json()
+
+        acr = self.create_random_name(prefix='acr', length=24)
+        image_source = "mcr.microsoft.com/k8se/quickstart:latest"
+        image_name = f"{acr}.azurecr.io/k8se/quickstart:latest"
+
+        acr_resource_id = self.cmd(f'acr create --sku basic -n {acr} -g {resource_group}').get_output_in_json()["id"]
+        self.cmd(f'acr import -n {acr} --source {image_source}')
+        roleAssignmentName = self.create_guid()
+        self.cmd(f'role assignment create --role acrpull --assignee {principal_id} --scope {acr_resource_id} --name {roleAssignmentName}')
+       
+        # wait for role assignment take effect
+        time.sleep(30)
+        
+        # Create CustomContainer SessionPool
+        sessionpool_name_custom = self.create_random_name(prefix='spcc', length=24)
+        ready_instances = 2
+        cpu = "0.5"
+        memory = "1Gi"
+        self.cmd(
+            f'containerapp sessionpool create -g {resource_group} -n {sessionpool_name_custom} -l {location} --container-type CustomContainer --environment {env_name} --ready-sessions {ready_instances} --image {image_name} --cpu {cpu} --memory {memory} --target-port 80 --registry-server {acr}.azurecr.io --registry-identity {user_identity_id} --mi-system-assigned',
+            checks=[
+                JMESPathCheck('name', sessionpool_name_custom),
+                JMESPathCheck('properties.containerType', "CustomContainer"),
```

Comment:
We should validate the provisioningState, identity type, the user identity, registry identity. 
For example:
You can add these Check and rerun your tests in playback mode without `--live`.
```
JMESPathCheck("properties.provisioningState", "Succeeded"),
JMESPathCheck("identity.type", "SystemAssigned, UserAssigned"),
JMESPathCheckExists(f'identity.userAssignedIdentities."{identity_rid_1}"'),
JMESPathCheck("length(properties.configuration.registries)", 1),
JMESPathCheck("properties.configuration.registries[0].identity", identity_rid_1),
JMESPathCheck("properties.configuration.registries[0].server", f'{acr}.azurecr.io'),
JMESPathCheck("properties.configuration.registries[0].username", ""),
JMESPathCheck("properties.configuration.registries[0].passwordSecretRef", ""),
```